core/domain/training_job_result_services.py

```diff
@@ -0,0 +1,43 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Functions for training job result domain object and protobuf message."""
+
+from core.classifiers import algorithm_registry
+from core.domain.proto import training_job_response_payload_pb2
+
+def get_proto_message_from_training_job_result(job_result):
```

Comment:
Maybe this should be part of the domain object itself since it's a conversion (similar to what we do in Oppia web backend for converting to/from a dict).

Reply:
That is how I implemented it earlier. However, I changed it to current structure after discussion with @seanlip. We agreed that we should have domain object and proto objects separate and service functions to convert one into other, much like, how domain object and storage layer works in Oppia. Domain objects don't have ability to store themselves, instead we have converter functions to convert one into the other.

Reply:
Protos themselves are data structures though, so it doesn’t feel like an exact analogy to domain and storage objects. I think it comparing json and domain objects is a bit closer.

Reply:
@seanlip, what do you think we should do here?

Reply:
If I'm reading this correctly, it doesn't touch an external storage layer, so I think you can implement it as training_job_result.to_proto_message(). The main requirement I have for domain objects is that we want to keep them self-contained and not dependent on external platform layers. If there's actually a storage layer call here then I'm misunderstanding something, so please correct me.

Reply:
I am reverting back to `to_proto` approach since there is no external platform layer call. Initially, I had implemented protos at platform layer which led to all confusion. Now, protos are part of service layer and hence we can go with @BenHenning's approach.